components/launcher/src/service.rs

```diff
@@ -95,8 +95,15 @@ where
     let mut reader = BufReader::new(out);
     let mut buffer = String::new();
     while reader.read_line(&mut buffer).unwrap() > 0 {
-        let line = output_format!(preamble &id, logkey "O", buffer);
-        writeln!(&mut io::stdout(), "{}", line).expect("unable to write to stdout");
+        let so = StructuredOutput::new(
```

Comment:
The panic should surface to the console or log and would likely be the last entry since it would crash the supervisor.

Reply:
It would log, but I'm pretty sure it *wouldn't* crash the supervisor. Just because a thread panics, doesn't mean the application crashes. That would only happen if the main thread were to try to join the panicked thread. Here's a toy example:
```rust
    let handle = std::thread::Builder::new()
        .spawn(move || panic!("I ded"))
        .unwrap();

    for _ in 1..5 {
        println!("I'm alive");
        std::thread::sleep(std::time::Duration::from_secs(5));
    }

    println!("calling join");
    handle.join().expect("join other thread");
    println!("join success");
```
When I run this locally, I only get a crash once I get to the `join` call:
```
➤ cargo run --release
I'm alive
thread '<unnamed>' panicked at 'I ded', src/main.rs:3:24
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
    …
I'm alive
I'm alive
I'm alive
calling join
thread 'main' panicked at 'join other thread: Any', src/libcore/result.rs:997:5
stack backtrace:
    …
```
I think an `error!` log would be more appropriate here. Panicking should be reserved for things that we think represent logical impossibilities.

Reply:
oh man I forgot that these are run in separate threads